### ISSUE TITLE
crashes on image analysis

### ISSUE THREAD

**Issue by lukepatrick** [wagoodman/dive#35]
on a mac, using brew to install dive `wagoodman/dive/dive: stable 0.0.5` . analyzing a very large centos image.

```
  Fetching image config...
  Building tree...
  Analyzing layers...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x139b3b6]

goroutine 1 [running]:
github.com/jroimartin/gocui.(*View).Clear(0x0)
	/home/wagoodman/go/src/github.com/jroimartin/gocui/view.go:392 +0x26
github.com/wagoodman/dive/ui.(*DetailsView).Render.func1(0xc467340000, 0x1, 0xc471d91858)
	/home/wagoodman/go/src/github.com/wagoodman/dive/ui/detailsview.go:119 +0x87
github.com/jroimartin/gocui.(*Gui).consumeevents(0xc467340000, 0xc471d919f0, 0x0)
	/home/wagoodman/go/src/github.com/jroimartin/gocui/gui.go:399 +0x17c
github.com/jroimartin/gocui.(*Gui).MainLoop(0xc467340000, 0x0, 0x0)
	/home/wagoodman/go/src/github.com/jroimartin/gocui/gui.go:381 +0x204
github.com/wagoodman/dive/ui.Run(0xc4203903c0, 0x39, 0x39, 0xc46a254400, 0x39, 0x40, 0x3fe14a3d82e9cd53, 0xc474fce000, 0x823e, 0x9800)
	/home/wagoodman/go/src/github.com/wagoodman/dive/ui/ui.go:330 +0x94c
github.com/wagoodman/dive/cmd.analyze(0x183cac0, 0xc42027bc60, 0x1, 0x1)
	/home/wagoodman/go/src/github.com/wagoodman/dive/cmd/analyze.go:22 +0x166
github.com/spf13/cobra.(*Command).execute(0x183cac0, 0xc42000a070, 0x1, 0x1, 0x183cac0, 0xc42000a070)
	/home/wagoodman/go/src/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/spf13/cobra.(*Command).ExecuteC(0x183cac0, 0x1, 0x1, 0x0)
	/home/wagoodman/go/src/github.com/spf13/cobra/command.go:852 +0x30a
github.com/spf13/cobra.(*Command).Execute(0x183cac0, 0xc42002a0b8, 0x0)
	/home/wagoodman/go/src/github.com/spf13/cobra/command.go:800 +0x2b
github.com/wagoodman/dive/cmd.Execute()
	/home/wagoodman/go/src/github.com/wagoodman/dive/cmd/root.go:28 +0x2d
main.main()
	/home/wagoodman/go/src/github.com/wagoodman/dive/main.go:34 +0x20
```

**Comment by wagoodman** [comment]
Thanks for the info! A few questions:

- What was the full command you ran? 
- Was this a public Docker image?
- Is this easily reproducible? or does it happen intermittently?

**Comment by lukepatrick** [comment]
1. `dive myimage:tag`
2. Kind of public, it's `FROM centos:7.5.1804 AS base`, we have created the Centos version of Stackstorm https://github.com/StackStorm/st2-docker/blob/master/images/stackstorm-all/Dockerfile
though our specific image is private due to adding in our own keys/security
3. This happens each time I try dive on this image. I can get dive to work without issue on some other much smaller images (nginx, redis, etc..)

**Comment by wagoodman** [comment]
@lukepatrick I haven't been able to reproduce it yet (with this image/dockerfile or other very large images, up to 2Gb). Though, it appears from the stack trace that one of the UI headers is not initialized before it is used.

- Can you download v0.0.8 and try this again to see if your issue is still there? 
- Do you see a drive.log entry that is odd?

I'm up for ideas on how to troubleshoot this further.

**Comment by lukepatrick** [comment]
With 0.0.8:
```
Analyzing Image
  Fetching metadata...



  Discovering layers... 56 %
  Discovering layers... 59 %
  Discovering layers... 71 %
  Discovering layers... 71 %
  Discovering layers... 79 %
  Discovering layers... 79 %
  Discovering layers... 82 %
  Discovering layers... 83 %
  Discovering layers... 83 %
  Discovering layers... 87 %
  Discovering layers... 87 %
  Discovering layers... 99 %
    ├─ fda91867fb6dfdf : [========================================>] 100 % (33832/33832)
    ├─ f4f26f4d70b34c7 : [========================================>] 100 % (3/3)
    ├─ f31e09f08ab4fd3 : [========================================>] 100 % (869/869)
    ├─ ee5611315e8144b : [========================================>] 100 % (4/4)
    ├─ ea02c0454b9352c : [========================================>] 100 % (703/703)
    ├─ ea018f49f0d9692 : [========================================>] 100 % (98/98)
    ├─ e0a5e1fec95928c : [========================================>] 100 % (3/3)
    ├─ d6d9892be6cecba : [========================================>] 100 % (128/128)
    ├─ d1dfafb3fa540ce : [========================================>] 100 % (6/6)
    ├─ c72f124ad6b0ea6 : [========================================>] 100 % (33224/33224)
    ├─ b2ddf7b5b35cd38 : [========================================>] 100 % (8888/8888)
    ├─ b273455e5bc6304 : [========================================>] 100 % (132/132)
    ├─ aa8f3b65002d39a : [========================================>] 100 % (12/12)
    ├─ aa0674dc13b2d68 : [========================================>] 100 % (530/530)
    ├─ a78fef1aba2e977 : [========================================>] 100 % (5/5)
    ├─ a6af82c7054a786 : [========================================>] 100 % (10673/10673)
    ├─ a52c936f36cdf41 : [========================================>] 100 % (4646/4646)
    ├─ a3016a5dc517086 : [========================================>] 100 % (7159/7159)
    ├─ 9ec6f2787fd950d : [========================================>] 100 % (3/3)
    ├─ 9d63a24c51e6070 : [========================================>] 100 % (173/173)
    ├─ 9ae55bc388d7e59 : [========================================>] 100 % (3/3)
    ├─ 9ad591e040697f9 : [========================================>] 100 % (12/12)
    ├─ 91fefc355a89c05 : [========================================>] 100 % (3/3)
    ├─ 89fa4c75e136d4b : [========================================>] 100 % (3/3)
    ├─ 81fdb7195eb209a : [========================================>] 100 % (20/20)
    ├─ 7df83ec3b933242 : [========================================>] 100 % (311/311)
    ├─ 74826e894f6b768 : [========================================>] 100 % (19/19)
    ├─ 7465a1bea325784 : [========================================>] 100 % (70/70)
    ├─ 73fbbe6ee3ad926 : [========================================>] 100 % (6/6)
    ├─ 738b55bbfecaa94 : [========================================>] 100 % (67/67)
    ├─ 6ccf474c1962d61 : [========================================>] 100 % (4710/4710)
    ├─ 6aa9374819b6504 : [========================================>] 100 % (1/1)
    ├─ 5dfe3c8f464166b : [========================================>] 100 % (3/3)
    ├─ 59d153cb2ad4d84 : [========================================>] 100 % (20/20)
    ├─ 5689f63094a6c02 : [========================================>] 100 % (3/3)
    ├─ 5473b4236fcf89a : [========================================>] 100 % (4/4)
    ├─ 52ef0c56aa1f7a4 : [========================================>] 100 % (3/3)
    ├─ 52526a2213873df : [========================================>] 100 % (3/3)
    ├─ 4e01631f60e7f75 : [========================================>] 100 % (7/7)
    ├─ 4c3b68a77175b82 : [========================================>] 100 % (3/3)
    ├─ 43ec6b00451ea80 : [========================================>] 100 % (3/3)
    ├─ 431b425a70e9f73 : [========================================>] 100 % (5/5)
    ├─ 416ecaa2c413524 : [========================================>] 100 % (10/10)
    ├─ 3f6092429c21191 : [========================================>] 100 % (4/4)
    ├─ 02bbaa1e78b3003 : [========================================>] 100 % (7209/7209)
  Fetching image config...
  Building tree...
  Analyzing layers...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x139b6c6]

goroutine 1 [running]:
github.com/jroimartin/gocui.(*View).Clear(0x0)
	/home/wagoodman/go/src/github.com/jroimartin/gocui/view.go:392 +0x26
github.com/wagoodman/dive/ui.(*DetailsView).Render.func1(0xc435bb6000, 0x1, 0xc487523838)
	/home/wagoodman/go/src/github.com/wagoodman/dive/ui/detailsview.go:119 +0x90
github.com/jroimartin/gocui.(*Gui).consumeevents(0xc435bb6000, 0xc4875239d0, 0x0)
	/home/wagoodman/go/src/github.com/jroimartin/gocui/gui.go:399 +0x17c
github.com/jroimartin/gocui.(*Gui).MainLoop(0xc435bb6000, 0x0, 0x0)
	/home/wagoodman/go/src/github.com/jroimartin/gocui/gui.go:381 +0x204
github.com/wagoodman/dive/ui.Run(0xc420096780, 0x39, 0x39, 0xc455db2000, 0x39, 0x40, 0x3fe14a3d82e9cd53, 0xc488282000, 0x823e, 0x9800)
	/home/wagoodman/go/src/github.com/wagoodman/dive/ui/ui.go:330 +0x94c
github.com/wagoodman/dive/cmd.analyze(0x183eb60, 0xc420295c90, 0x1, 0x1)
	/home/wagoodman/go/src/github.com/wagoodman/dive/cmd/analyze.go:37 +0x183
github.com/spf13/cobra.(*Command).execute(0x183eb60, 0xc4200ae010, 0x1, 0x1, 0x183eb60, 0xc4200ae010)
	/home/wagoodman/go/src/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/spf13/cobra.(*Command).ExecuteC(0x183eb60, 0x2, 0x19256c8, 0x0)
	/home/wagoodman/go/src/github.com/spf13/cobra/command.go:852 +0x30a
github.com/spf13/cobra.(*Command).Execute(0x183eb60, 0xc4201d1f78, 0xc4201d1f58)
	/home/wagoodman/go/src/github.com/spf13/cobra/command.go:800 +0x2b
github.com/wagoodman/dive/cmd.Execute()
	/home/wagoodman/go/src/github.com/wagoodman/dive/cmd/root.go:29 +0x2d
main.main()
	/home/wagoodman/go/src/github.com/wagoodman/dive/main.go:38 +0xad
```

```cat dive.log 
level=debug msg="Starting Dive..."
level=debug msg="Starting Dive..."
```

**Comment by wagoodman** [comment]
Thanks for the extra info! That's just what I needed to reproduce it. The trick is not that the image is large in size (bytes) but instead your image has more layers than what can fit on the screen. It's fixed in e7a05d4b358b3ae55ae3302dcd465b1979b73f62 and will make it into v0.1.0.
